engine/apps/public_api/views/shift_swap.py

```diff
@@ -0,0 +1,98 @@
+import logging
+
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.exceptions import NotFound
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.serializers import BaseSerializer
+
+from apps.api.permissions import AuthenticatedRequest
+from apps.api.views.shift_swap import BaseShiftSwapViewSet
+from apps.auth_token.auth import ApiTokenAuthentication
+from apps.public_api.throttlers.user_throttle import UserThrottle
+from apps.schedules.models import ShiftSwapRequest
+from apps.user_management.models import User
+from common.api_helpers.custom_fields import TimeZoneAwareDatetimeField
+from common.api_helpers.exceptions import BadRequest
+from common.api_helpers.mixins import RateLimitHeadersMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ShiftSwapViewSet(RateLimitHeadersMixin, BaseShiftSwapViewSet):
+    # set authentication and permission classes
+    authentication_classes = (ApiTokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    # public API customizations
+    throttle_classes = [UserThrottle]
+
+    def get_queryset(self):
+        schedule_id = self.request.query_params.get("schedule_id", None)
+        beneficiary = self.request.query_params.get("beneficiary", None)
+        benefactor = self.request.query_params.get("benefactor", None)
+        starting_after = self.request.query_params.get("starting_after", None)
+        untaken = self.request.query_params.get("untaken", "false") == "true"
```

Comment:
nit: maybe `taken` would work better here? `untaken=False` double negations can be hard on the brain 😄